### ISSUE TITLE
regression: don't start services every few ms if condition fails

### ISSUE THREAD

**Issue by evverx** [systemd/systemd#3167]
Again https://github.com/systemd/systemd/issues/2467

```
# cat <<EOF > /etc/systemd/system/test.socket
[Socket]
ListenStream=/run/test.ctl
EOF
# cat <<EOF > /etc/systemd/system/test.service
[Unit]
Requires=test.socket
ConditionPathExistsGlob=/tmp/nonexistent

[Service]
ExecStart=/bin/true
EOF
# systemctl daemon-reload

# systemctl start test.socket

# echo a | nc -U /run/test.ctl & # hangs

# journalctl -f -u test
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Trying to enqueue job test.service/start/replace
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Installed new job test.service/start as 192
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Enqueued job test.service/start as 192
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: ConditionPathExistsGlob=/tmp/nonexistent failed.
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Starting requested but condition failed. Not starting unit.
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Job test.service/start finished, result=done
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Trying to enqueue job test.service/start/replace
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Installed new job test.service/start as 244
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Enqueued job test.service/start as 244
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: ConditionPathExistsGlob=/tmp/nonexistent failed.
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Starting requested but condition failed. Not starting unit.
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Job test.service/start finished, result=done
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Trying to enqueue job test.service/start/replace
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Installed new job test.service/start as 296
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Enqueued job test.service/start as 296
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: ConditionPathExistsGlob=/tmp/nonexistent failed.
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Starting requested but condition failed. Not starting unit.
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Job test.service/start finished, result=done
May 01 20:50:07 systemd-testsuite systemd[1]: test.service: Trying to enqueue job test.service/start/replace
...
and so on
```

This regression was introduced by 7629ec4642b03517742

```
TEST RUN: https://github.com/systemd/systemd/issues/2467 [FAILED]
Makefile:10: recipe for target 'run' failed
make: *** [run] Error 1
7629ec4642b03517742d09b7303c204fddf82108 is the first bad commit
commit 7629ec4642b03517742d09b7303c204fddf82108
```

~~The test is coming soon.~~ See https://github.com/systemd/systemd/pull/3168

**Comment by poettering** [comment]
Well, TriggerLimitIntervalSec= and TriggerLimitIntervalBurst= default to 2500 activations per 5s now. Until that limit is hit we'll keep on trying to activate the service unit. Are you saying that this doesn't work?

**Comment by poettering** [comment]
So, I ran the lines you pasted above, and what I get is this in the logs:

```
May 02 12:16:50 fedora-cntnr systemd[1]: test.socket: Trigger limit hit, refusing further activation.
May 02 12:16:50 fedora-cntnr systemd[1]: test.socket: Unit entered failed state.
```

And the echo/nc line aborted a short delay with:

```
Ncat: Connection reset by peer.
```

Hence, to me this looks like the right things happened: it tried to activate the thing a 2500 times, in a tight loop, then hit the ratelimit and gave up.

systemctl status says this btw:

```
# systemctl status test.socket
● test.socket
   Loaded: loaded (/etc/systemd/system/test.socket; static; vendor preset: disabled)
   Active: failed (Result: trigger-limit-hit) since Mon 2016-05-02 12:16:50 CEST; 2min 13s ago
   Listen: /run/test.ctl (Stream)

May 02 12:16:41 fedora-cntnr systemd[1]: Listening on test.socket.
May 02 12:16:50 fedora-cntnr systemd[1]: test.socket: Trigger limit hit, refusing further activation.
May 02 12:16:50 fedora-cntnr systemd[1]: test.socket: Unit entered failed state.
```

Hence, looks pretty good to me, even explains with the "trigger-limit-hit" result what actually happened.

**Comment by evverx** [comment]
> Well, TriggerLimitIntervalSec= and TriggerLimitIntervalBurst= default to 2500 activations per 5s now. Until that limit is hit we'll keep on trying to activate the service unit. Are you saying that this doesn't work?

It depends:)

`nspawn`-container:

```
# journalctl -b -u test.socket --no-hostname --no-pager
-- Logs begin at Tue 2016-05-03 01:32:34 UTC, end at Tue 2016-05-03 01:34:50 UTC. --
May 03 01:34:25 systemd[1]: Listening on test.socket.
May 03 01:34:50 systemd[1]: test.socket: Trigger limit hit, refusing further activation.
May 03 01:34:50 systemd[1]: test.socket: Unit entered failed state.
```

`QEMU`:

```
May 03 01:40:40 systemd[1]: test.socket: Incoming traffic
May 03 01:40:40 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:40 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:40 systemd[1]: test.socket: Incoming traffic
May 03 01:40:40 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:40 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:40 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
May 03 01:40:41 systemd[1]: test.socket: Incoming traffic
May 03 01:40:41 systemd[1]: test.socket: Changed listening -> running
May 03 01:40:41 systemd[1]: test.socket: Changed running -> listening
and so on
```

(I've bisected the "right" value for that VM: `TriggerLimitBurst=66`)

**Comment by poettering** [comment]
@evverx hmm, you are saying the default rate limit settings are simply too high to take effect on that qemu? Yuck...

so for your qemu 66 connections per 5s is when it starts to get hit? That really low I must say. Irritatingly low in fact...

I wonder if we maybe should pick different defaults based on Accept=yes/no. Specifically, pick a much lower value for Accept=no than for Accept=yes, after all Accept=no means there is a singleton service, and if that lives that shortly something is really off... But for Accept=yes a busy loop isn't that likely anyway, since PID1 already accepted the connection and thus we know that the traffic has been processed and doesn't stay queued.

Hence, what do you say if we change the defaults to maybe:

TriggerLimitIntervalSec=2s in both cases
but
TriggerLimitBurst=20 on Accept=no
and
TriggerLimitBurst=200 on Accept=yes

Or something like that? Does that sound appropriate?

**Comment by evverx** [comment]
> I wonder if we maybe should pick different defaults based on Accept=yes/no. Specifically, pick a much lower value for Accept=no than for Accept=yes

I like this idea.

> Does that sound appropriate?

Yeah, `TriggerLimitBurst=20` on `Accept=no` and `TriggerLimitBurst=200` on `Accept=yes` looks reasonable to me.

**Comment by poettering** [comment]
suggested timeout default change fix waiting in PR #3198.
